READY: Fix BitcoinTransactionid class

basvijzendoorn opened the pull request:
Added conversions in BitcoinTransactionid class and fixed tests to handle txid correctly from [commit d04c509](https://github.com/devos50/tribler/commit/d04c5096fee1c80321536e7fc2f9c86d7f90e8aa).

ghost commented:
BitcoinTransactionid is a string representation and therefore doesn't need number conversions. Working on a fix.

ghost commented:
Fix applied. Working on other tests.

basvijzendoorn committed:
Fix BitcoinTransactionid class

Added conversions in BitcoinTransactionid class and fixed tests to handle txid correctly.

ghost commented:
Fix other tests with MultiChain or Bitcoin Payments.

devos50 commented:
Good! Please let me know when it's done.

Also, please prepend `WIP` in the title if you're working on it and `READY` when it's ready for review 👍

ghost commented:
It is `READY`

devos50 commented on Tribler/community/market/core/bitcoin_transaction_id.py:
```diff
@@ -16,3 +16,18 @@ def __init__(self, transaction_id):
+        return not self.__eq__(other)
+
+    def __hash__(self):
+        return hash(self._transaction_id)
```
Please modify your Python IDE that it automatically adds a newline at the end of the file when you save it.

basvijzendoorn replied:
Done

devos50 reviewed (changes requested):
Other than that minor comment, looks fine!

Please fix the comment and I will merge it.

ghost commented:
Fixed Changes
